public/app/features/browse-dashboards/components/BrowseActions/utils.ts

```diff
@@ -26,3 +39,269 @@ export function buildBreakdownString(
+  summary: {
+    total: number;
+    successCount: number;
+    failedCount: number;
+  };
+}
+
+interface FulfilledResource {
+  uid: string;
+  data: ResourceWrapper;
+  title: string;
+}
+
+/**
+ * Execute bulk move operation for selected dashboards
+ * TODO: Follow up https://github.com/grafana/git-ui-sync-project/issues/315#issuecomment-3075920997 for better approach
+ */
+export const bulkMoveResources = async ({
+  selectedItems,
+  targetFolderPath,
+  repository,
+  mutations,
+  options,
+}: BulkMoveRequest): Promise<BulkMoveResult> => {
+  // 1. Get dashboard data for all selected dashboards
+  const dashboardsToMove = Object.keys(selectedItems.dashboard).filter((uid) => selectedItems.dashboard[uid]);
+
+  const dashboardDataResults = await Promise.allSettled(
+    dashboardsToMove.map(async (uid) => {
+      try {
+        const dto = await getDashboardAPI().getDashboardDTO(uid);
+        const sourcePath =
+          'meta' in dto
+            ? dto.meta.k8s?.annotations?.[AnnoKeySourcePath] || dto.meta.provisionedExternalId
+            : dto.metadata?.annotations?.[AnnoKeySourcePath];
+
+        if (!sourcePath) {
+          throw new Error(`No source path found for dashboard ${uid}`);
+        }
+
+        const baseUrl = `/apis/provisioning.grafana.app/v0alpha1/namespaces/${config.namespace}`;
+        const url = `${baseUrl}/repositories/${repository.name}/files/${sourcePath}`;
+        const fileResponse = await getBackendSrv().get(url);
+
+        return {
+          uid,
+          data: fileResponse,
+          title: 'dashboard' in dto ? dto.dashboard.title : dto.spec?.title || 'Unknown',
+        };
+      } catch (error) {
+        console.error(`Failed to get dashboard data for ${uid}:`, error);
+        throw error;
+      }
+    })
+  );
+
+  // 2. Filter successful data fetches
+  const { fulfilledResources, dataFetchFailures } = dashboardDataResults.reduce<{
+    fulfilledResources: FulfilledResource[];
+    dataFetchFailures: MoveResultFailed[];
+  }>(
+    (acc, result, index) => {
+      if (result.status === 'fulfilled') {
+        acc.fulfilledResources.push(result.value);
+      } else {
+        acc.dataFetchFailures.push({
+          uid: dashboardsToMove[index],
+          status: 'failed',
+          failureType: 'data-fetch',
+        });
+      }
+      return acc;
+    },
+    {
+      fulfilledResources: [],
+      dataFetchFailures: [],
+    }
+  );
+
+  // 3. Execute moves for dashboards with valid data
+  // In the executeBulkMove function, replace the move logic:
+  const moveResults = await Promise.allSettled(
+    fulfilledResources.map(async ({ uid, data, title }): Promise<MoveResultSuccess | MoveResultFailed> => {
+      try {
+        const fileName = formatFileName(data);
+        const newPath = formatFilePath(targetFolderPath, fileName);
+        const body = data.resource.file;
+
+        if (!body) {
+          throw new Error(`No file content found for dashboard ${uid}`);
```

Comment:
nitpick for the errors throughout here - they aren't marked up for translation, but I don't know how these are being surfaced, personally :D